commit 86604e2bd536ea48832dd0bf3d95b15de4de2733
Author: Steve Dickson <steved@redhat.com>
Date:   Thu Sep 6 10:22:11 2018 -0400

    statd: fix use-after-free in monitor list if insertion fails
    
    If nsm_insert_monitored_host() fails while saving the record to
    stable storage, we can't just assume the entry was new. Existing
    records must be removed from the list before being freed.
    
    Reviewed-by: Chuck Lever <chuck.lever@oracle.com>
    Signed-off-by: Frank Sorenson <sorenson@redhat.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/statd/monitor.c b/utils/statd/monitor.c
index 45c4346..9400048 100644
--- a/utils/statd/monitor.c
+++ b/utils/statd/monitor.c
@@ -197,7 +197,7 @@ sm_mon_1_svc(struct mon *argp, struct svc_req *rqstp)
 
 	if (!nsm_insert_monitored_host(dnsname,
 				(struct sockaddr *)(char *)&my_addr, argp)) {
-		nlist_free(NULL, clnt);
+		nlist_free(existing ? &rtnl : NULL, clnt);
 		goto failure;
 	}
 
